File: leet-code/455. Assign Cookies/AssignCookies455.java
package com.example.algorithm;

import java.util.Arrays;

public class AssignCookies455 {

    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);
        int answer = 0, child = 0, cookie = 0;
        while (child < g.length && cookie < s.length) {
            if (g[child] <= s[cookie]) {
                answer++;
                child++;
            }
            cookie++;
        }
        return answer;
    }
}

File: leet-code/455. Assign Cookies/AssignCookies455.kt
package com.example

class AssignCookies455 {
    fun findContentChildren(g: IntArray, s: IntArray): Int {
        g.sort()
        s.sort()
        var maxNumber = 0
        var idx = 0
        for (size in s) {
            if (idx >= g.size) {
                break
            }
            if (size >= g[idx]) {
                maxNumber += 1
                idx += 1
            }
        }
        return maxNumber
    }
}
